Проект

MargaritaGorshneva2-MD-25 opened the pull request.

MargaritaGorshneva2-MD-25 committed:
Add files via upload

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,8 @@
+def driver():
+    driver = webdriver.Chrome()
+    driver.maximize_window() # или другие настройки, если нужны
+    yield driver
```
Нужно исправить: закрытие драйвера должно происходить в рамках этой фикстуры.

ShamilovR commented on locators.py:
```diff
@@ -0,0 +1,48 @@
+from selenium.webdriver.common.by import By
+
+base_url = "https://stellarburgers.nomoreparties.site/"
```
Можно лучше: здесь и далее: урлы лучше хранить в отдельном модуле, т.к. к локаторам они отношения не имеют

ShamilovR commented on tests/test_constructor_tabs.py:
```diff
@@ -0,0 +1,37 @@
+from locators import Locators
+import time
+
+base_url = "https://stellarburgers.nomoreparties.site/"
+
+
+@pytest.fixture(scope="function")
+def driver():
+    driver = webdriver.Chrome()
+    driver.maximize_window()
+    yield driver
```
Нужно исправить: здесь и далее: тестовые модули должны содержать только тестовые классы и тестовые сценарии.

ShamilovR commented on tests/test_constructor_tabs.py:
```diff
@@ -0,0 +1,37 @@
+    # Клик по вкладке ингредиента
+    tab_element = driver.find_element(By.XPATH, locator)
+    tab_element.click()
+    time.sleep(1) # небольшая задержка, чтобы дать время для скролла. Возможно, потребуется adjust
```
Нужно исправить: здесь и далее: sleep в тестах плохая практика, т.к. для загрузки страницы/элемента может потребоваться меньше или больше времени. Для ожиданий нужно использовать модуль expected_condition.

ShamilovR commented on tests/test_login.py:
```diff
@@ -0,0 +1,62 @@
+
+
+def test_login_from_main_page():
+    driver = webdriver.Chrome()
```
Нужно исправить: здесь и далее: инициализация/закрытие драйвера должна происходить в рамках фикстур. Такая реализация позволяет избежать дублирования кода.

ShamilovR commented on tests/test_login.py:
```diff
@@ -0,0 +1,62 @@
+def test_login_from_main_page():
+    driver = webdriver.Chrome()
+    driver.maximize_window()
+    try:
```
Нужно исправить: здесь и далее: использование try/except для тестовых сценариев недопустимо. Их использование говорит о нестабильности тестового сценария.

ShamilovR commented on tests/test_login.py:
```diff
@@ -0,0 +1,62 @@
+        driver.find_element(By.XPATH, Locators.login_button_main_page).click()
+        check_login_page_opened(driver) # Вызов вспомогательной функции
+    finally:
+        driver.quit()
```
Нужно исправить: здесь и далее: тесовый сценарий должен заканчиваться проверкой (assert`ом)

ShamilovR commented on tests/test_logout.py:
```diff
@@ -0,0 +1,36 @@
+    try:
+        WebDriverWait(driver, 5).until(EC.presence_of_element_located((By.XPATH, Locators.login_button_main_page))) # Проверяем наличие кнопки "Войти в аккаунт"
+        # Или, если перенаправление на другую страницу:
+        # WebDriverWait(driver, 5).until(EC.url_to_be(base_url + "login")) # Если перенаправляет на страницу логина
```
Нужно исправить: код не должен содержать закомментированный код

ShamilovR commented on tests/test_registration.py:
```diff
@@ -0,0 +1,50 @@
+def test_successful_registration(driver):
+    driver.get(base_url + "register")
+    driver.find_element(By.XPATH, Locators.name_field).send_keys("Имя")
+    driver.find_element(By.XPATH, Locators.email_field).send_keys("email@example.com")
```
Нужно исправить: при использовании захардкоженного email тест пройдет только первый раз, т.к. функционал не позволяет регистрировать пользователей с одинаковым email . Нужно использовать рандомную генерацию email для каждого теста.

ShamilovR commented on tests/test_constructor_tabs.py:
```diff
@@ -0,0 +1,37 @@
+    driver.get(base_url)
+
+    # Получаем начальное положение раздела до скролла (не обязательно, но может быть полезно для отладки)
+    section_locator = Locators.get_section_locator(tab_name)
+    initial_location = driver.find_element(By.XPATH, section_locator).location
+
+    # Клик по вкладке ингредиента
+    tab_element = driver.find_element(By.XPATH, locator)
+    tab_element.click()
+    time.sleep(1) # небольшая задержка, чтобы дать время для скролла. Возможно, потребуется adjust
+
+    # Проверка, что раздел проскроллился
+    # (текущее положение раздела отличается от начального)
+    final_location = driver.find_element(By.XPATH, section_locator).location
+    assert initial_location != final_location, f"Раздел '{tab_name}' не проскроллился"
```
Нужно исправить: проверка расположения элементов может быть недостоверной. Для корректной проверки переключения табов нужно использовать другой способ, например, проверка изменения атрибута class у элемента.

MargaritaGorshneva2-MD-25 committed:
Исправление

ShamilovR commented on tests/test_constructor_tabs.py:
```diff
@@ -0,0 +1,29 @@
+    ("Соусы", Locators.SAUCES_TAB),
+    ("Начинки", Locators.FILLINGS_TAB),
+])
+def test_ingredient_tab_switch(driver, tab_name, locator):
```
Нужно исправить: здесь и далее: тестовые сценарии должны быть оформлены в тестовых классах и тестовых методах. Такая реализация позволяет объединять несколько сценариев в сьюты в рамках одного модуля.

ShamilovR commented on tests/test_personal_area.py:
```diff
@@ -0,0 +1,63 @@
+    check_personal_area_page_opened(driver)
+
+
+def check_personal_area_page_opened(driver):
```
Нужно исправить: здесь и далее: тестовые модули должны содержать только тестовые классы и тестовые сценарии.

ShamilovR commented on tests/test_personal_area.py:
```diff
@@ -0,0 +1,63 @@
+    )
+
+    login_button.click()
+    check_personal_area_page_opened(driver)
```
Нужно исправить: здесь и далее: тесовый сценарий должен заканчиваться проверкой (assert`ом)

MargaritaGorshneva2-MD-25 committed:
Доработка

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,19 @@
+    driver.quit()
+
+# Проверка перехода на страницу профиля
+def check_profile_page_opened(driver):
+    WebDriverWait(driver, 10).until(EC.url_to_be(BASE_URL + "account/profile"))
+    assert driver.find_element(*Locators.LOGOUT_BUTTON).is_displayed(), "Страница профиля не открылась."
```
Нужно исправить: модуль conftest предназначен только для фикстур. Подобные функции лучше вынести в helpers

ShamilovR commented on tests/test_login.py:
```diff
@@ -0,0 +1,22 @@
+    (Locators.LOGIN_LINK_REGISTRATION, BASE_URL + "login"),  # Для регистрации
+    (Locators.LOGIN_LINK_FORGOT_PASSWORD, BASE_URL + "login"),  # Для восстановления пароля
+])
+def test_login_redirection(driver, locator, expected_url):
```
Нужно исправить: здесь и далее: тестовые сценарии должны быть оформлены в тестовых классах и тестовых методах. Такая реализация позволяет объединять несколько сценариев в сьюты в рамках одного модуля. 
Прошу обратить внимание на примечание "здесь и далее"

ShamilovR commented on tests/test_registration.py:
```diff
@@ -0,0 +1,74 @@
+import string
+
+# Функция для генерации случайного email
+def generate_random_email():
```
Нужно исправить: тестовые модули должны содержать только тестовые классы и тестовые сценарии.

MargaritaGorshneva2-MD-25 committed:
Доработка и появления файла helpest

ShamilovR commented on tests/test_personal_area.py:
```diff
@@ -0,0 +1,53 @@
+
+
+class TestPersonalArea:
+    def login_and_go_to_profile(self, driver, email, password):
```
Нужно исправить: здесь и далее: тестовые модули должны содержать только тестовые классы и тестовые сценарии.

ShamilovR commented on tests/test_login.py:
```diff
@@ -0,0 +1,33 @@
+        (Locators.LOGIN_LINK_FORGOT_PASSWORD_FORM, BASE_URL + "login"), # Изменили локатор
+    ])
+    def test_redirection(self, driver, locator, expected_url): # Изменили имя тестового метода
+        if locator in (Locators.LOGIN_LINK_REGISTRATION_FORM, Locators.LOGIN_LINK_FORGOT_PASSWORD_FORM):
+            # Для перехода на страницы регистрации и восстановления пароля
+            # нужно сначала перейти на страницу входа или регистрации
+            driver.get(BASE_URL + "register") # Можно использовать "login" или "register" в зависимости от вашего приложения
+
+            if locator == Locators.LOGIN_LINK_FORGOT_PASSWORD_FORM:
+                # Если проверяем ссылку "Восстановить пароль", нужно сначала перейти на страницу входа
+                 WebDriverWait(driver, 10).until(EC.presence_of_element_located(Locators.LOGIN_LINK_REGISTRATION_FORM)).click()
+                 WebDriverWait(driver, 10).until(EC.url_to_be(BASE_URL + "login")) #Проверка, что мы на странице /login
```
Нужно исправить: if, else в тестовых сценариях недопустимы, т.к. условия предполагают разные варианты исхода. А значит это должны быть два отдельных тестовых сценария

MargaritaGorshneva2-MD-25 committed:
Снова

MargaritaGorshneva2-MD-25 committed:
Доработка номер не помню

ShamilovR commented on tests/test_logout.py:
```diff
@@ -0,0 +1,37 @@
+            EC.element_to_be_clickable(Locators.LOGIN_BUTTON)
+        ).click()
+
+        WebDriverWait(driver, 10).until(EC.url_contains(BASE_URL + "account/profile"))
```
Нужно исправить: здесь и далее: тесовый сценарий должен заканчиваться проверкой (assert`ом)

MargaritaGorshneva2-MD-25 committed:
~

Ayantea1 commented on tests/test_personal_area.py:
```diff
@@ -0,0 +1,32 @@
+    def _wait_for_element(self, driver, locator, timeout=10):
+        return WebDriverWait(driver, timeout).until(EC.presence_of_element_located(locator))
+
+    def _login(self, driver): # Используем данные из config.py напрямую
```
Повторяющийся код лучше выносить в фикстуры

Ayantea1 commented on tests/test_registration.py:
```diff
@@ -0,0 +1,43 @@
+        current_url = driver.current_url
+        self._wait_for_element(driver, Locators.REGISTER_BUTTON).click()
+
+        try:
```
Нужно исправить: в тесте не должна быть ожидаема ошибка, связанная с невозможностью исполнить код. Переформулируй тест

Ayantea1 commented on tests/test_registration.py:
```diff
@@ -0,0 +1,43 @@
+
+    def test_successful_registration(self, driver):
+        driver.get(BASE_URL + "register")
+        self._fill_registration_form(driver, "Имя", TEST_EMAIL, TEST_PASSWORD)  # Используем TEST_PASSWORD
```
Нужно исправить: при наличии захардкорженных данных, успешной будет только первая регистрация

MargaritaGorshneva2-MD-25 committed:
Доработка после перехода

Ayantea1 commented on helpest.py:
```diff
@@ -0,0 +1,46 @@
+
+
+def generate_random_email():
+    username = ''.join(random.choices(string.ascii_lowercase, k=10))
```
Можно улучшить: можно использовать специальную библиотеку для создания тестовых данных, например Faker

Ayantea1 commented on conftest.py:
```diff
@@ -0,0 +1,26 @@
+from locators import Locators
+from URL import BASE_URL
+from config import TEST_EMAIL, TEST_PASSWORD
+from helpers import wait_for_element
```
Нужно исправить здесь и далее: название модуля в импорте указано неверно

Ayantea1 commented on tests/test_constructor_tabs.py:
```diff
@@ -0,0 +1,29 @@
+        ("Соусы", Locators.SAUCES_TAB, Locators.SAUCES_SECTION, "tab_active"),
+        ("Начинки", Locators.FILLINGS_TAB, Locators.FILLINGS_SECTION, "tab_active"),
+    ])
+    def test_ingredient_tab_switching(self, driver, tab_name, locator, expected_section_locator, expected_active_class):
```
Нужно исправить здесь и далее : фикстура драйвера не была импортирована, при запуске она не находится

Ayantea1 replied:
Нужно исправить: тест не начинается, поскольку нет захода на страницу

Ayantea1 commented on conftest.py:
```diff
@@ -0,0 +1,26 @@
+@pytest.fixture(scope="function")
+def driver():
+    chrome_options = Options()
+    driver = webdriver.Chrome(options=chrome_options)
```
Нужно исправить: тут имеет смысл не только открывать браузер, но и заходить на стартовую страницу.
Либо нужно указывать этот вход в каждом тесте. Сейчас в ряде тестов у тебя просто открывается браузер

Ayantea1 commented on tests/test_logout.py:
```diff
@@ -0,0 +1,21 @@
+        logout_button.click()
+
+        WebDriverWait(logged_in_user, 10).until(EC.url_to_be(BASE_URL + "login")) # Ожидаем перенаправления на страницу логина
+        assert logged_in_user.current_url == BASE_URL + "login", "Перенаправление на страницу входа не выполнено после выхода"
```
Нужно исправить: logged_in_user не импортирован, как и driver

Ayantea1 commented on conftest.py:
```diff
@@ -0,0 +1,26 @@
+@pytest.fixture(scope="function")
+def logged_in_user(driver):
+    driver.get(BASE_URL + "login")
+    wait_for_element(driver, Locators.EMAIL_FIELD).send_keys(TEST_EMAIL)
```
Нужно исправить здесь и далее: wait_for_element ждет, но не возвращает веб элемент. Заполнение полей не происходит

MargaritaGorshneva2-MD-25 committed:
Ещё доработка

Да, заметила ошибку в импортах, но не до конца поняла что ты имеешь в виду под проблемами с взаимодействием с элементами. Возможно у меня уже просто замылился взгляд.

MargaritaGorshneva2-MD-25 committed:
Очень сильные изменения

Я постаралась исправить всё, но к сожалению очень часто программа ругается на локаторы. Я уже не знаю как это исправить.
